bin/kano-sync

```diff
@@ -67,58 +69,77 @@ def do_login():
     login_success, value = login_using_token()
     if not login_success:
         logger.debug('token login not successful msg: {}'.format(value))
-        if args.silent:
+        if silent:
             logger.debug('exiting because of --silent')
-            sys.exit()
+            return False
         else:
             logger.debug('trying to log in with kano-login')
             run_cmd(dir_path + '/bin/kano-login')
             login_success, value = login_using_token()
             if not login_success:
-                display_msg('Login not possible, error: ' + value, 'error')
-                sys.exit(1)
+                display_msg(
+                    'Login not possible, error: {} error'.format(value),
+                    'error',
+                    dialog=dialog
+                )
+                return False
+    return True
+
 
 def _any_notifications_available(profile):
     return ('notifications' in profile and
             type(profile['notifications']) is list and
             len(profile['notifications']) > 0)
 
-def do_sync():
+
+def _display_notifications():
+    profile = load_profile()
+    if _any_notifications_available(profile):
+        for n in profile['notifications']:
+            image = n["image"] if "image" in n else None
+            command = n["command"] if "command" in n else None
+            sound = n["sound"] if "sound" in n else None
+            display_generic_notification(
+                n["title"], n["byline"], image, command, sound, "world"
+            )
+
+            mark_notification_read(n['id'])
+
+        profile['last_notifications_prompt'] = int(time.time())
+        save_profile(profile, skip_kdesk_refresh=True)
+    else:
+        logger.debug('No notifications to show')
```

Comment:
Maybe you don't like early exits but this would flow more linearly if it were

```python
def _display_notifications():
    profile = load_profile()
    if not _any_notifications_available(profile):
        logger.debug('No notifications to show')
        return

    for n in profile['notifications']:
        ...
```